src/audio/pipeline/pipeline-graph.c

```diff
@@ -201,10 +203,14 @@ int pipeline_connect(struct comp_dev *comp, struct comp_buffer *buffer,
 	 * overwriting the modified header. FIXME: this is still a problem with
 	 * different cores.
 	 */
-	if (!list_is_empty(comp_list))
-		dcache_writeback_invalidate_region(uncache_to_cache(comp_list->next),
-						   sizeof(struct list_item));
+	further_buffers_exist = !list_is_empty(comp_list);
+	needs_sync = uncache_to_cache(comp_list->next);
+	if (further_buffers_exist)
+		dcache_writeback_region(needs_sync, sizeof(struct list_item));
```

Comment:
This will WB the contents around list_item as we only WB/INV on cache line alignment and in multiples of cache line size.

Reply:
@lgirdwood sure, that's exactly what's needed. We need to sync the list head and we cannot avoid syncing the respective cache line, and that's ok